lib/rouge/lexers/red.rb

```diff
@@ -0,0 +1,196 @@
+          integer! issue! lit-path! lit-word! logic! map! native! none! object!
+          op! pair! paren! path! percent! point! refinement! routine! set-path!
+          set-word! string! tag! time! tuple! typeset! unset! url! vector! word!
+        )
+      end
+
+      def self.operators
+        @operators ||= Set.new %w(
+          + < <> % << = * / - > <= >= ** // == =? >> >>> and or
+        )
+      end
+
+      state :root do
+        # comments
+        rule /;.+$/, Comment::Single
+        rule /comment {.*}/m, Comment::Multiline
+
+        rule /[+-]?[0-9]+\.[0-9]*E[+-]?[0-9]+/, Literal::Number::Float
+        rule /[+-]?\.[0-9]+E[+-]?[0-9]+/, Literal::Number::Float
+        rule /[+-]?[0-9]+E[+-]?[0-9]+/, Literal::Number::Float
+        rule /[+-]?[0-9]*\.[0-9]+?/, Literal::Number::Float
+        rule /[+-]?[0-9]+/, Literal::Number::Integer
+
+        rule /\./, Punctuation
+        rule /,/, Punctuation
+        rule /;/, Punctuation
+        rule /\(/, Punctuation
+        rule /\)/, Punctuation
+        rule /\{/, Punctuation
+        rule /\}/, Punctuation
+        rule /\[/, Punctuation
+        rule /\]/, Punctuation
+        rule /\^\^/, Punctuation
+
+        rule /<[^>]*>/, Name::Label
+        rule /\s+/, Text::Whitespace
+        rule /""".*?"""/m, Literal::String
+        rule /"([^"\\]|\\.)*"/, Literal::String
+        rule /'''.*?'''/m, Literal::String
+        rule /'([^'\\]|\\.)*'/, Literal::String
+
+        # email
+        rule /\w[-\w._]*\@\w[-\w._]*/, Keyword::Type
+
+        # time
+        rule  /\b\d+:\d+(:\d+)?/, Keyword::Type
+
+        # url
+        rule  /\w[-\w_]*\:(\/\/)?\w[-\w._]*(:\d+)?/, Keyword::Type
+
+        # date
+        rule /(\b\d{1,4}[-\/]\d{1,2}[-\/]\d{1,2}|\d{1,2}[-\/]\d{1,2}[-\/]\d{1,4})\b/, Keyword::Type
+
+        # tuple
+        rule  /\b\d{1,3}\.\d{1,3}\.\d{1,3}(\.\d{1,3}){0,9}/, Keyword::Type
+
+        # pair
+        rule  /[+-]?\d+x[-+]?\d+/, Keyword::Type
+
+        # binary -- ToDo
+
+        # issue
+        rule /#\w[-\w'*.]*/, Keyword::Type
+
+        # file
+        rule /%[-\w\.\/]+/, Keyword::Type
+
+        # char
+        rule /#"(\^[-@\/_~^"HKLM\[]|.)"/, Literal::String::Char
+
+        # get word
+        rule  /\:\w[-\w'*.?!]*/, Name::Property
+
+        # set word
+        rule  /\w[-\w'*.?!]*\:/, Name::Variable
+
+        # lit word
+        rule /'\w[-\w'*.?!]*/, Name::Entity
+
+        # word
+        rule /\b\w+[-\w'*.!?]*/, Name
+
+        # get path
+        rule  /:\w[-\w'*.?!]*(\/\w[-\w'*.?!]*)(\/\w[-\w'*.?!]*)*/, Name::Variable
+
+        # set path
+        rule  /\w[-\w'*.?!]*(\/\w[-\w'*.?!]*)(\/\w[-\w'*.?!]*)*:/, Name::Variable
+
+        # lit path
+        rule /'\w[-\w'*.?!]*(\/\w[-\w'*.?!]*)(\/\w[-\w'*.?!]*)*/, Name::Variable
+
+        # path
+        rule  /\w[-\w'*.?!]*(\/\w[-\w'*.?!]*)(\/\w[-\w'*.?!]*)*/, Name::Variable
+
+        # refinement
+        rule  /\/\w[-\w'*.?!]*/, Name::Variable
+
+        rule id do |m|
```

Comment:
Is this the only place that `id` is used? If so, it can probably simply be inlined here for clarity.

Reply:
I can do that.